src/solve/mirk.jl

```diff
@@ -258,7 +258,7 @@ end
 
 @views function __mirk_loss(u, p, y, pt::StandardBVProblem, bc::BC, mesh, cache) where {BC}
     y_ = recursive_unflatten!(y, u)
-    soly_ = VectorOfArray(y_)
+    soly_ = MIRKSol(y_, mesh, cache)
```

Comment:
Just pass the ODESolution itself. You've already constructed it and are updated the values?

Reply:
Yeah, I just created a simple solution type here, since what we need here is only the interpolation of this solution, so it's probably not necessary to create a new `ODESolution` every time in loss function?

Reply:
Just using ODESolution here while we are evaluating boundary conditions